platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/serialize/DefaultClassEncoder.kt

```diff
@@ -138,16 +140,24 @@ class DefaultClassEncoder(
     }
 
     private
-    fun ensureClassCanBeFoundInGradleRuntimeClassLoader(className: String, originalClassLoader: ClassLoader) {
+    fun WriteContext.ensureClassCanBeFoundInGradleRuntimeClassLoader(className: String, originalClassLoader: ClassLoader) {
         try {
             classForName(className, gradleRuntimeClassLoader)
         } catch (e: ClassNotFoundException) {
-            throw ConfigurationCacheError(
-                "Class '${className}' cannot be encoded because ${describeClassLoader(originalClassLoader)} could not be encoded " +
-                    "and the class is not available through the default class loader.\n" +
-                    scopeLookup.describeKnownClassLoaders() +
-                    "Please report this error, run './gradlew --stop' and try again.",
-                e
+            onProblem(
+                PropertyProblem(
+                    trace,
+                    StructuredMessage.build {
+                        text("Class ")
+                        reference(className)
+                        text(" cannot be encoded because ")
+                        text(describeClassLoader(originalClassLoader))
```

Comment:
This method could either use brackets around the set of classloaders, or return "none" instead of an empty string.

Currently, it produces a message like this:

```
- Gradle runtime: Class 'org.gradle.internal.cc.impl.ProjectWithWork' cannot be encoded because Gradle runtime class loader 'jdk.internal.loader.ClassLoaders$AppClassLoader@7c53a9eb' of type 'jdk.internal.loader.ClassLoaders$AppClassLoader' could not be encoded and the class is not available through the default class loader.
These are the known class loaders:
	- ScriptClassLoader(groovy-script-/Users/rafael/sources/gradle/trees/tree14/platforms/core-configuration/configuration-cache/build/tmp/teŝt files/Configurati.Test/ciqc6/build.gradle-loader)
Please report this error, run './gradlew --stop' and try again.
- Gradle runtime: Class 'org.gradle.internal.cc.impl.fingerprint.ConfigurationCacheFingerprint$EnvironmentVariablesPrefixedBy' cannot be encoded because Gradle runtime class loader 'jdk.internal.loader.ClassLoaders$AppClassLoader@7c53a9eb' of type 'jdk.internal.loader.ClassLoaders$AppClassLoader' could not be encoded and the class is not available through the default class loader.
These are the known class loaders:

Please report this error, run './gradlew --stop' and try again.
- Gradle runtime: Class 'org.gradle.internal.cc.impl.fingerprint.ConfigurationCacheFingerprint$GradleEnvironment' cannot be encoded because Gradle runtime class loader 'jdk.internal.loader.ClassLoaders$AppClassLoader@7c53a9eb' of type 'jdk.internal.loader.ClassLoaders$AppClassLoader' could not be encoded and the class is not available through the default class loader.
These are the known class loaders:

Please report this error, run './gradlew --stop' and try again.
```

Reply:
Well spotted, @abstratt, thanks!